### PR TITLE
[Acknowledgements] Reactify

### PR DESCRIPTION
**Pull request by zaliqarosli**
### Brief summary of changes
Reactify acknowledgements module.

Certain hard coded data in the previous version of the module was directly transferred over to this reactification without fixing. This PR only reactifies.

screenshot 2019-03-05 17 57 07

### CODE REVIEW

File: modules/acknowledgements/jsx/acknowledgementsIndex.js [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+   *
+   * @return {object}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
```
Comment (maltheism) [inline modules/acknowledgements/jsx/acknowledgementsIndex.js]: Should we remove console.error?

File: modules/acknowledgements/jsx/acknowledgementsIndex.js [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+  }
+
+  /**
+   * Store the value of the element in this.state.formData
+   *
+   * @param {string} formElement - name of the form element
+   * @param {string} value - value of the form element
+   */
+  setFormData(formElement, value) {
+    const formData = this.state.formData;
```
Comment (maltheism) [inline modules/acknowledgements/jsx/acknowledgementsIndex.js]: @HenriRabalais pointed out to me in another PR how assigning state to a variable like this isn't good. You could do this instead:
`const formData = Object.assign({}, this.state.formData);`

File: modules/acknowledgements/jsx/acknowledgementsIndex.js [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+  handleSubmit(e) {
+    let formData = this.state.formData;
+    let formObject = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== '') {
+        formObject.append(key, formData[key]);
+      }
+    }
+    formObject.append('fire_away', 'Add');
+    $.ajax({
```
Comment (maltheism) [inline modules/acknowledgements/jsx/acknowledgementsIndex.js]: We're using fetch now instead of jquery ajax.

File: modules/acknowledgements/jsx/acknowledgementsIndex.js [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {object} row - row content indexed by column
+   *
+   * @return {*} a formated table cell for a given column
+   */
+  formatColumn(column, cell, row) {
+    return <td>{cell}</td>;
```
Comment (maltheism) [inline modules/acknowledgements/jsx/acknowledgementsIndex.js]: Should we always wrap or leave as is?
```
return (
<td>{cell}</td>
);
```

Comment (HenriRabalais) [reply modules/acknowledgements/jsx/acknowledgementsIndex.js]: I personally prefer inline formatting when possible.

Comment (maltheism) [reply modules/acknowledgements/jsx/acknowledgementsIndex.js]: @HenriRabalais Okay, we can save two lines. This reminds me of me not using curly braces when only a single line of code after a for loop or an if statement. Took me awhile to break that habit.

File: modules/acknowledgements/jsx/acknowledgementsIndex.js (hunk 1 above)
Comment (zaliqarosli) [reply modules/acknowledgements/jsx/acknowledgementsIndex.js]: im not sure. @HenriRabalais should we be?

Comment (HenriRabalais) [reply modules/acknowledgements/jsx/acknowledgementsIndex.js]: Not really sure. It's more a useful too for testing to have this hear, but as @maltheism pointed it, our jsx linter doesn't like it. I don't really have an opinion either way.

Comment (zaliqarosli) [reply modules/acknowledgements/jsx/acknowledgementsIndex.js]: i think the jsx linter on major was changed? the console logs for .error passes on the branch (or has been in previously merged PRs) so i think i'll keep it!

**Comment by johnsaigle** [comment]
Failing Travis because of PHPCS errors in the acknowledgements module

**Comment by zaliqarosli** [comment]
@kongtiaowang i skipped tests like in previous PRs. let me know if this is okay, and remove "I don't understand Travis" label if it is. thanks!

**Comment by kongtiaowang** [comment]
Screen Shot 2019-03-04 at 10 52 55 AM

**Comment by kongtiaowang** [comment]
Add button is missing, when the table is empty.
Screen Shot 2019-04-01 at 10 46 25 AM

File: modules/acknowledgements/jsx/acknowledgementsIndex.js [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      formData: {},
+      error: false,
+      isLoaded: false,
+      affiliationsOptions: {
+        douglas: 'Douglas',
```
Comment (kongtiaowang) [inline modules/acknowledgements/jsx/acknowledgementsIndex.js]: Hard coding?

File: modules/acknowledgements/jsx/acknowledgementsIndex.js [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+        mcgill: 'McGill',
+      },
+      degreesOptions: {
+        bachelors: 'Bachelors',
```
Comment (kongtiaowang) [inline modules/acknowledgements/jsx/acknowledgementsIndex.js]: Hard coding?

**Comment by kongtiaowang** [comment]
If a user clicks the Save button several times. It will input the save value several times.

File: modules/acknowledgements/jsx/acknowledgementsIndex.js (hunk 5 above)
Comment (zaliqarosli) [reply modules/acknowledgements/jsx/acknowledgementsIndex.js]: yep, these were hardcoded in the module before reactification. the module doesn't currently have configurations options / tables in the back

**Comment by zaliqarosli** [comment]
https://github.com/aces/Loris/pull/4438

**Comment by zaliqarosli** [comment]
thanks for the approval @kongtiaowang ! this PR will be blocked by #4438 until it gets merged

**Comment by maltheism** [comment]
Hi @zaliqarosli I pulled your PR and everything works. I'll approve once the other PR is merged.

**Comment by zaliqarosli** [comment]
https://github.com/aces/Loris/pull/4438 merged
